GCC 12.1 compilation errors

Compiling with a gcc 12.1 shows the next errors:

```
In file included from /home/rnou/gekkofs/external/hermes/include/hermes/buffer.hpp:18,
[build]                  from /home/rnou/gekkofs/external/hermes/include/hermes/exposed_memory.hpp:19,
[build]                  from /home/rnou/gekkofs/external/hermes/include/hermes/async_engine.hpp:28,
[build]                  from /home/rnou/gekkofs/external/hermes/include/hermes.hpp:4,
[build]                  from /home/rnou/gekkofs/include/client/preload_context.hpp:33,
[build]                  from /home/rnou/gekkofs/include/client/preload.hpp:33,
[build]                  from /home/rnou/gekkofs/src/client/hooks.cpp:31:
[build] /home/rnou/gekkofs/external/hermes/include/hermes/logging.hpp:182:18: error: field 'm_log_callbacks' has incomplete type 'hermes::log::logger::callback_set' {aka 'std::array<std::function<void(const std::__cxx11::basic_string<char>&, hermes::log::level, int, const std::__cxx11::basic_string<char>&, const std::__cxx11::basic_string<char>&, int)>, 6>'}
[build]   182 |     callback_set m_log_callbacks;
[build]       |                  ^~~~~~~~~~~~~~~
[build] In file included from /usr/include/c++/12.1.0/bits/hashtable_policy.h:34,
[build]                  from /usr/include/c++/12.1.0/bits/hashtable.h:35,
[build]                  from /usr/include/c++/12.1.0/unordered_map:46,
[build]                  from /home/rnou/gekkofs/external/hermes/include/hermes/async_engine.hpp:6:
[build] /usr/include/c++/12.1.0/tuple:1595:45: note: declaration of 'using callback_set = struct std::array<std::function<void(const std::__cxx11::basic_string<char>&, hermes::log::level, int, const std::__cxx11::basic_string<char>&, const std::__cxx11::basic_string<char>&, int)>, 6>' {aka 'struct std::array<std::function<void(const std::__cxx11::basic_string<char>&, hermes::log::level, int, const std::__cxx11::basic_string<char>&, const std::__cxx11::basic_string<char>&, int)>, 6>'}
[build]  1595 |   template<typename _Tp, size_t _Nm> struct array;
[build]       |       
```


It can be solved including `#include <array>` in async_engine.hpp

--- a/include/hermes/async_engine.hpp
+++ b/include/hermes/async_engine.hpp
@@ -15,6 +15,7 @@
 #include <chrono>
 #include <algorithm>
 #include <cstring>
+#include <array>
 
 // C includes
 #include <mercury.h>

